src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -471,6 +479,30 @@ export class SavedObjectsClient {
     });
   }
 
+  /**
+   * Opens a Point In Time (PIT) against the indices for the specified Saved Object types.
+   * The returned `id` can then be passed to `SavedObjects.find` to search against that PIT.
+   *
+   * @param {string|Array<string>} type
+   * @param {object} [options] - {@link SavedObjectsPointInTimeOptions}
+   * @property {string} [options.keepAlive]
+   * @property {string} [options.preference]
+   * @returns {promise} - { id: string }
+   */
+  public openPointInTimeForType = (
+    type: string | string[],
+    { keepAlive, preference }: SavedObjectsOpenPointInTimeOptions = {}
+  ): ReturnType<SavedObjectsApi['openPointInTimeForType']> => {
+    const types = Array.isArray(type) ? type.join(',') : type;
+    return this.savedObjectsFetch(this.getPath([types, '_pit']), {
```

Comment:
Was wondering the same thing. For our use case with SO exports, that's already going through the `_export` route anyway, so unless we see other teams needing this on the client I'm not sure if it's truly necessary.

Reply:
Yea, I have the same feeling. Unless we already foresee some need to have this exposed on the client-side, I would keep that exclusively server-side for now. (but let's wait for @rudolf's opinion before reverting these changes 😅 )

Reply:
We're adding PIT searches so that plugins can page through more than 10k saved objects, so although it might work to load all saved objects into the browser today, this will probably change soon.

So the only use case I see is for external integrations with the HTTP API (we shouldn't expose this on the public saved objects client). 

1. Should we give users the ability to page through >10k so's?
   Probably. It kinda sucks if you integrate with the API and then it stops working when you reach 10k objects. The users with the largest deployments are probably more likely to have API integrations.
2. Is this a denial of service risk?
   The API would be open to all authenticated users and opening long-standing PIT's can be resource intensive. So this is probably a risk, but then again a user searching over 2 years worth of log data is also going to impact an ES cluster. But maybe we should not expose this to all users by default.

So I kind of see this as a nice to have, but if the permissions discussion below holds this back I think we can proceed without it.

Reply:
> Should we give users the ability to page through >10k so's?

On the client side, the `PIT` would not allow to 'page' results, as we would not be able to switch to an arbitrary page or to get back to a previous one. So I don't really see any obvious usages on the client side anyway?

Reply:
I'll edit my reply to clarify, but I meant users using the API (I agree we don't need anything on the client-side)